### PR TITLE
[HACK Week] Adjust Zendesk contact information setup

### LINKED ISSUE #8488: [Support Requests] Support contact name is not getting stored
The Support contact information dialog that's used to fill the user data before submitting a Support Request ticket contains two fields: email and name, right now, the name field is not getting properly stored and triggered with the ticket creation itself.

https://user-images.githubusercontent.com/5920403/223303786-01e2772c-0cc6-400a-8dc5-f6fd27145a02.mp4

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix issue #8488 by refactoring how the ZendeskHelper class configures the Contact email and name and fixing the bug where the user name always shows up empty in the contact card dialog.

How to Test
==========
### Scenario 1
1. Reinstall the app and go to the Help & Support section of the app inside the settings view
2. Select the Contact section and fill in both the email and name information fields
3. Select the Contact field again and verify that both email and name are displayed correctly

### Scenario 2
1. Reinstall the app and go to the Help & Support section of the app inside the settings view
2. Select the Contact support section and verify if the dialog requesting the email and name fields appears
3. Fill in both information and wait for the Support Request form to appear, once inside there, go back to the Help & Support view
4. Click the Contact section and verify if both email and name are displayed correctly

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/ZendeskHelper.kt [hunk 1]
```diff
@@ -70,8 +75,29 @@ class ZendeskHelper(
      * supportEmail and reset their identity on logout, we need to ensure that the correct identity is set all times.
      * Check [requireIdentity], [refreshIdentity] & [clearIdentity] for more details about how Zendesk identity works.
      */
-    private var supportEmail: String? = null
-    private var supportName: String? = null
+    var supportEmail: String? = null
+        get() = AppPrefs.getSupportEmail()
+            .takeIf { it.isNotEmpty() }
+            ?: supportHelper.getSupportEmailSuggestion(accountStore.account, selectedSite.getIfExists())
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/support/ZendeskHelper.kt:83]: Do you think this is a good candidate to create an AccountRepository / SupportAccountRepository? A class that can use different data sources to get the user's email and name

**Comment by atorresveiga** [comment]
Great work @ThomazFB!
I only found a small issue. When there is no user information, the Contact email section keeps showing `not set` even after I set the contact information. Here is a video.

Steps to reproduce:
1. Fresh install the app ( clear app data should work too )
2. Go to the Help & Support section of the app inside the settings view
3. Check that the Contact section description displays `not set`
4. Tap the Contact section
5. Fill in both the email and name information fields
6. Check that the Contact section description keeps displaying `not set` even after we set the fields

https://user-images.githubusercontent.com/18119390/224059425-0a0d0b1a-9bf3-4e62-9495-1e858c69b71b.mp4

Other than that, the rest of the test cases are working great

**Comment by ThomazFB** [comment]
Nice catch @atorresveiga! Thanks for that! I thought I had introduced a regression to the view and started investigating what I did, but it turns out this problem was already in trunk. So I fixed it by properly triggering a data refresh after we submit the contact info. Fixed here: https://github.com/woocommerce/woocommerce-android/pull/8489/commits/b066c527702eeb96efa8b0aa4d33852e6041adb1

Let me know how that goes!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/support/ZendeskHelper.kt (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/support/ZendeskHelper.kt:83]: I agree, good point! Maybe we can discuss an opportunity to introduce something like this in [this PR](https://github.com/woocommerce/woocommerce-android/pull/8512) that I'll be moving to review today. I divided the ZendeskHelper there in two different types: `ZendeskSettings` and `ZendeskManager`, the Settings one is concentrating all configuration-related code, which includes the Identity info. Maybe we can explore an option like your suggestion there.

Anyway, I suggest we move this suggestion to that PR as it's almost ready for review!

**Review by atorresveiga — APPROVED** [review]
Thanks for working on this, Thomaz!
There is a conflict with one of the PR files. Feel free to merge this PR once that conflict gets resolved.
LGTM! :shipit:

**Pull request merged**